test/Common/MoistThermodynamics/runtests.jl

```diff
@@ -124,215 +124,261 @@ using LinearAlgebra
 end
 
 
-@testset "moist thermodynamics - constructor consistency" begin
-  # Make sure `ThermodynamicState` arguments are returned unchanged
+@testset "moist thermodynamics - default behavior accuracy" begin
+  # Input arguments should be accurate within machine precision
+  # Temperature is approximated via saturation adjustment, and should be within a physical tolerance
 
-  # FIXME/TODO: Increasing `n` to 40 results in broken tests
-  # due to failure in saturation adjustment. This needs
-  # to be improved/made more robust.
-  n = 35
   FT = Float64
-  θ_liq_ice = collect(range(250,400,length=n))
-  T = collect(range(250,400,length=n))
-  ρ = collect(range(0.5,2.0,length=n))
-  q_tot_dry = zeros(n)
-  q_tot_moist = collect(range(0.0,0.2,length=n))
-  q_liq = collect(range(0.0,0.1,length=n))
-  q_ice = collect(range(0.0,0.05,length=n))
-
-  e_int = collect(range(10_000.0,200_000.0,length=n))
-  p = collect(range(10_000.0,200_000.0,length=n))
-  q_pt_dry = PhasePartition.(q_tot_dry, q_liq .* 0, q_ice .* 0)
-  q_pt_moist = PhasePartition.(q_tot_moist, q_liq, q_ice)
+  e_int, ρ, q_tot, q_pt, T, p, θ_liq_ice = MT.tested_convergence_range(FT, 50)
+  tol = 1e-1
+
+  # PhaseEquil
+  ts_exact = PhaseEquil.(e_int, ρ, q_tot, FT(1e-6), 10000, MT.saturation_adjustment_SecantMethod)
+  ts = PhaseEquil.(e_int, ρ, q_tot)
+  # Should be machine accurate:
+  @test all(getproperty.(PhasePartition.(ts),:tot) .≈ getproperty.(PhasePartition.(ts_exact),:tot))
+  @test all(internal_energy.(ts) .≈ internal_energy.(ts_exact))
+  @test all(air_density.(ts) .≈ air_density.(ts_exact))
+  # Approximate:
+  @test max(abs.(air_pressure.(ts)    .- air_pressure.(ts_exact))./air_pressure.(ts_exact)...) < tol
```

Comment:
Would be good to distinguish `rtol` and `atol` here. What you have here is relative tolerance `rtol`. But it seems to me `tol` is an absolute tolerance, no?